pkg/jsonapi/resource.go

```diff
@@ -240,6 +240,30 @@ func (r *Resource) SaveAsMultipart(fields []string) error {
 	return nil
 }
 
+/*
+Delete a resource from the server. Response is empty on success
+*/
+func (r *Resource) Delete() error {
+	url := r.Links.Self
+	if url == "" {
+		// Make an extra effort
+		url = fmt.Sprintf("/%s/%s", r.Type, r.Id)
+	}
+	_, err := r.API.request("DELETE", url, nil, "")
+
+	if err != nil {
+		var e *RedirectError
+		if errors.As(err, &e) {
+			r.Redirect = e.Location
+			return nil
```

Comment:
This is copied from `.Reload()`, right? I'm not opposed, but I don't believe it makes sense to want to handle a redirect during delete. I think a simple reraising of the error will suffice.